### PR TITLE
CRNS-136/ChannelList

### PR DESCRIPTION
**Pull request by DanC5**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/components/ChannelList/hooks/usePaginatedChannels.ts [hunk 1]
```diff
@@ -78,6 +110,7 @@ export const usePaginatedChannels = ({
 
   const loadNextPage = () => {
     if (hasNextPage) return queryChannels();
+    return null;
   };
```
Comment (virdesai) [inline src/components/ChannelList/hooks/usePaginatedChannels.ts:114]:
```suggestion
  const loadNextPage = () => (hasNextPage ? queryChannels() : null);
```

File: src/components/ChannelList/hooks/usePaginatedChannels.ts [hunk 2]
```diff
@@ -1,28 +1,60 @@
...
+  Channel,
+  ChannelFilters,
+  ChannelOptions,
+  ChannelSort,
+  LiteralStringForUnion,
+  UnknownType,
+} from 'stream-chat';
 
 import { MAX_QUERY_CHANNELS_LIMIT } from '../utils';
+import { useChatContext } from '../../../contexts/chatContext/ChatContext';
```
Comment (virdesai) [inline src/components/ChannelList/hooks/usePaginatedChannels.ts:12]:
```suggestion

import { useChatContext } from '../../../contexts/chatContext/ChatContext';
```

File: src/components/ChannelList/hooks/listeners/useChannelTruncated.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+> = {
+  setChannels: React.Dispatch<
+    React.SetStateAction<Channel<At, Ch, Co, Ev, Me, Re, Us>[]>
+  >;
+  setForceUpdate: React.Dispatch<React.SetStateAction<number>>;
+  onChannelTruncated?: (
+    setChannels: React.Dispatch<
+      React.SetStateAction<Channel<At, Ch, Co, Ev, Me, Re, Us>[]>
+    >,
+    e: Event<At, Ch, Co, Ev, Me, Re, Us>,
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelTruncated.ts:28]:
```suggestion
    event: Event<At, Ch, Co, Ev, Me, Re, Us>,
```

File: src/components/ChannelList/hooks/listeners/useChannelTruncated.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  Us extends UnknownType = UnknownType
+>({
+  onChannelTruncated,
+  setChannels,
+  setForceUpdate,
+}: Parameters<At, Ch, Co, Ev, Me, Re, Us>) => {
+  const { client } = useChatContext<At, Ch, Co, Ev, Me, Re, Us>();
+
+  useEffect(() => {
+    const handleEvent = (e: Event<At, Ch, Co, Ev, Me, Re, Us>) => {
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelTruncated.ts:48]:
```suggestion
    const handleEvent = (event: Event<At, Ch, Co, Ev, Me, Re, Us>) => {
```

File: src/components/ChannelList/hooks/listeners/useChannelTruncated.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      if (typeof onChannelTruncated === 'function') {
+        onChannelTruncated(setChannels, e);
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelTruncated.ts:50]:
```suggestion
        onChannelTruncated(setChannels, event);
```

File: src/components/ChannelList/hooks/listeners/useChannelHidden.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  Us extends UnknownType = UnknownType
+> = {
+  setChannels: React.Dispatch<
+    React.SetStateAction<Channel<At, Ch, Co, Ev, Me, Re, Us>[]>
+  >;
+  onChannelHidden?: (
+    setChannels: React.Dispatch<
+      React.SetStateAction<Channel<At, Ch, Co, Ev, Me, Re, Us>[]>
+    >,
+    e: Event<At, Ch, Co, Ev, Me, Re, Us>,
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelHidden.ts:27]:
```suggestion
    event: Event<At, Ch, Co, Ev, Me, Re, Us>,
```

File: src/components/ChannelList/hooks/listeners/useChannelHidden.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  Re extends UnknownType = UnknownType,
+  Us extends UnknownType = UnknownType
+>({
+  onChannelHidden,
+  setChannels,
+}: Parameters<At, Ch, Co, Ev, Me, Re, Us>) => {
+  const { client } = useChatContext<At, Ch, Co, Ev, Me, Re, Us>();
+
+  useEffect(() => {
+    const handleEvent = (e: Event<At, Ch, Co, Ev, Me, Re, Us>) => {
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelHidden.ts:46]:
```suggestion
    const handleEvent = (event: Event<At, Ch, Co, Ev, Me, Re, Us>) => {
```

File: src/components/ChannelList/hooks/listeners/useChannelHidden.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+      if (typeof onChannelHidden === 'function') {
+        onChannelHidden(setChannels, e);
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelHidden.ts:48]:
```suggestion
        onChannelHidden(setChannels, event);
```

File: src/components/ChannelList/hooks/listeners/useChannelHidden.ts [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+      } else {
+        setChannels((channels) => {
+          const index = channels.findIndex(
+            (c) => c.cid === (e.cid || e.channel?.cid),
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelHidden.ts:52]:
```suggestion
            (channel) => channel.cid === (event.cid || event.channel?.cid),
```

File: src/components/ChannelList/hooks/listeners/useChannelDeleted.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  Us extends UnknownType = UnknownType
+> = {
+  setChannels: React.Dispatch<
+    React.SetStateAction<Channel<At, Ch, Co, Ev, Me, Re, Us>[]>
+  >;
+  onChannelDeleted?: (
+    setChannels: React.Dispatch<
+      React.SetStateAction<Channel<At, Ch, Co, Ev, Me, Re, Us>[]>
+    >,
+    e: Event<At, Ch, Co, Ev, Me, Re, Us>,
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelDeleted.ts:27]:
```suggestion
    event: Event<At, Ch, Co, Ev, Me, Re, Us>,
```

File: src/components/ChannelList/hooks/listeners/useChannelDeleted.ts [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  Re extends UnknownType = UnknownType,
+  Us extends UnknownType = UnknownType
+>({
+  onChannelDeleted,
+  setChannels,
+}: Parameters<At, Ch, Co, Ev, Me, Re, Us>) => {
+  const { client } = useChatContext<At, Ch, Co, Ev, Me, Re, Us>();
+
+  useEffect(() => {
+    const handleEvent = (e: Event<At, Ch, Co, Ev, Me, Re, Us>) => {
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelDeleted.ts:46]:
```suggestion
    const handleEvent = (event: Event<At, Ch, Co, Ev, Me, Re, Us>) => {
```

File: src/components/ChannelList/hooks/listeners/useChannelDeleted.ts [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+      if (typeof onChannelDeleted === 'function') {
+        onChannelDeleted(setChannels, e);
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelDeleted.ts:48]:
```suggestion
        onChannelDeleted(setChannels, event);
```

File: src/components/ChannelList/hooks/listeners/useChannelDeleted.ts [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+      } else {
+        setChannels((channels) => {
+          const index = channels.findIndex(
+            (c) => c.cid === (e.cid || e.channel?.cid),
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useChannelDeleted.ts:52]:
```suggestion
            (channel) => channel.cid === (event.cid || event.channel?.cid),
```

File: src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+  Us extends UnknownType = UnknownType
+> = {
+  setChannels: React.Dispatch<
+    React.SetStateAction<Channel<At, Ch, Co, Ev, Me, Re, Us>[]>
+  >;
+  onAddedToChannel?: (
+    setChannels: React.Dispatch<
+      React.SetStateAction<Channel<At, Ch, Co, Ev, Me, Re, Us>[]>
+    >,
+    e: Event<At, Ch, Co, Ev, Me, Re, Us>,
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts:29]:
```suggestion
    event: Event<At, Ch, Co, Ev, Me, Re, Us>,
```

File: src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+  Re extends UnknownType = UnknownType,
+  Us extends UnknownType = UnknownType
+>({
+  onAddedToChannel,
+  setChannels,
+}: Parameters<At, Ch, Co, Ev, Me, Re, Us>) => {
+  const { client } = useChatContext<At, Ch, Co, Ev, Me, Re, Us>();
+
+  useEffect(() => {
+    const handleEvent = async (e: Event<At, Ch, Co, Ev, Me, Re, Us>) => {
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts:48]:
```suggestion
    const handleEvent = async (event: Event<At, Ch, Co, Ev, Me, Re, Us>) => {
```

File: src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+      if (typeof onAddedToChannel === 'function') {
+        onAddedToChannel(setChannels, e);
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts:50]:
```suggestion
        onAddedToChannel(setChannels, event);
```

File: src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+      } else {
+        if (e.channel?.id && e.channel?.type) {
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts:52]:
```suggestion
        if (event.channel?.id && event.channel?.type) {
```

File: src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+          const channel = await getChannel<At, Ch, Co, Ev, Me, Re, Us>({
+            client,
+            id: e.channel.id,
+            type: e.channel.type,
```
Comment (virdesai) [inline src/components/ChannelList/hooks/listeners/useAddedToChannelNotification.ts:56]:
```suggestion
            id: event.channel.id,
            type: event.channel.type,
```

File: src/components/ChannelList/ChannelList.tsx [hunk 19]
```diff
@@ -194,79 +222,142 @@ ChannelList.propTypes = {
...
+    onRemovedFromChannel,
+    setChannels,
+  });
+
+  useUserPresence<At, Ch, Co, Ev, Me, Re, Us>({
+    setChannels,
+  });
+
+  return (
+    <List<At, Ch, Co, Ev, Me, Re, Us>
```
Comment (virdesai) [inline src/components/ChannelList/ChannelList.tsx:340]: oh interesting

File: src/components/ChannelList/ChannelList.tsx [hunk 20]
```diff
@@ -194,79 +222,142 @@ ChannelList.propTypes = {
...
+    onChannelDeleted,
+    onChannelHidden,
+    onChannelTruncated,
+    onChannelUpdated,
+    onMessageNew,
+    onRemovedFromChannel,
+    onSelect,
+    options = {},
+    setFlatListRef,
+    sort = {},
```
Comment (virdesai) [inline src/components/ChannelList/ChannelList.tsx:267]: this not something that abc's?

Comment (DanC5) [reply src/components/ChannelList/ChannelList.tsx:267]: this is a holdover from when we were doing caps first, on the others been doing caps in line with the others, so will update this

File: src/components/ChannelList/ChannelList.tsx (hunk 19 above)
Comment (DanC5) [reply src/components/ChannelList/ChannelList.tsx:340]: yeah don't love that syntax

**Review by virdesai — APPROVED** [review]

**Pull request merged**